frontend/src/pages/UserPage.jsx

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from 'react'
+import userService from '../services/users'
+import UserInfo from '../components/UserInfo' 
+
+const UserPage = ({setDefaultRole}) => {
+  const [user, setUser] = useState({})
+
+  useEffect(() => {
+    userService.get().then(user => setUser(user))
+  }, [])
+
+  const handleRoleSwitch = () => {
+    const newRole = switchRole();
+    userService.setRole(newRole).then(_ => {
+      setUser({...user, role: newRole})
+      localStorage.setItem('role', newRole);
+      setDefaultRole(newRole)
+    })  }
+
+  const switchRole = () => {
+    if (user.role == "grower") {
+      return "retailer"
+    } else {
+      return "grower"
+    }
+  }
```

Comment:
```suggestion
  const switchRole = () => (user.role === 'grower' ? 'retailer' : 'grower')
```

Reply:
This version is a bit more concise and avoids the if-else block. However, your solution works perfectly too, I just personally prefer the cleaner look of this one, haha!